src/mrinufft/operators/off_resonance.py

```diff
@@ -0,0 +1,330 @@
+"""Off Resonance correction Operator wrapper.
+
+Based on the implementation of Guillaume Daval-Frérot in pysap-mri:
+https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/operators/fourier/orc_wrapper.py
+"""
+
+import math
+import numpy as np
+
+from .._utils import get_array_module
+
+from .base import FourierOperatorBase, CUPY_AVAILABLE, AUTOGRAD_AVAILABLE
+from .interfaces.utils import is_cuda_array
+
+if CUPY_AVAILABLE:
+    import cupy as cp
+
+if AUTOGRAD_AVAILABLE:
+    import torch
+
+
+def get_interpolators_from_fieldmap(
+    fieldmap, readout_time, n_time_segments=6, n_bins=(40, 10), mask=None
+):
+    r"""Create B and C matrices to approximate ``exp(-2j*pi*fieldmap*t)``.
+
+    Here, B has shape ``(n_time_segments, len(t))``
+    and C has shape ``(n_time_segments, *fieldmap.shape)``.
+
+    From Sigpy: https://github.com/mikgroup/sigpy
+    and MIRT (mri_exp_approx.m): https://web.eecs.umich.edu/~fessler/code/
+
+    Parameters
+    ----------
+    fieldmap : np.ndarray
+        Rate map defined as ``fieldmap = R2*_map + 1j * B0_map``.
```

Comment:
I wonder, isn't there a world where we want to correct only based on `R2*_map` ? The current parameters don't allow the user to do it simply as if no imaginary B0 is given, the R2* map will be assumed to be B0. So why not just require explicitly R2*_map and B0_map separately ?

Reply:
You are right, I have now refactored the function which now accepts separate B0 and R2* (also easier for the user, which does not have to manually create the complex field map.